presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -458,7 +459,7 @@ private PlanNode createRemoteMaterializedExchange(ExchangeNode exchange, Rewrite
                     ImmutableList.copyOf(symbolToColumnMap.values()),
                     Optional.of(partitioningMetadata));
 
-            TableScanNode scan = createTemporaryTableScan(
+            TemporaryTableScanNode scan = createTemporaryTableScan(
```

Comment:
Why we need a TemporaryTableScanNode? Is it for bypassing the classLoader problem :)? 
Can we have following in ConnectorContext:
```
//ConnectorMetadataAccess will wrap around and provide controlled access to ConnectorMetadata.
ConnectorMetadataAccess getConnectorMetadataAccess(ConnectorId connectorId);

Then we can get tableHandle + columnHandle + ConnectorTableLayoutHandle that will be needed for query federation case.
We have to make sure we check the access control though. 
```